### ISSUE TITLE
Iterator usage Kata appears to have unintended errors

### ISSUE THREAD

**Issue by mattmazzola** [tddbin/es6katas.org#19]
If i understand correctly, the TDD is setup in a way that assumes the class provided is correct, but it isn't so the user is struggling to find problems with changes to the tests when in fact it's the class's implementation. By fixing the class this allows the users to focus on *usage* as the Kata is titled.

1st problem:
this logic of using `this.users.length > 0` indicates when there is any left, which is the opposite of when the iterator is done.

2nd problem:
The array shift method modifies the array within the next() call which means that for the call of anyLeft the array will already be changed and prematurely determine the iterator is done.
In other words:
The iterator will incorrectly output `{value: 'user: Bob', done: true}` because the `this.users.length > 0`

I'm sure there is an easier way to solve this but here is what I ended up using.
Possible solution:
```
class ConsumableUsers {
  constructor() {
    this.users = ['Alice', 'Bob'];
    this.nextIndex = 0;
  }
  get nextUser() {
    if (this.nextIndex < this.users.length) {
      return `user: ${this.users[this.nextIndex]}`;
    }
    return void 0;
  }
  get anyLeft() {
    var isDone = this.nextIndex >= this.users.length;
    this.nextIndex++;
    return isDone;
  }
}
```

**Comment by wolframkriesing** [comment]
Hi mattmazola,

this kata might be a bit more complicated, I just did it again too (I dont remember all the solutions either :)). But following the instructions leads (at least me) towards the right solution, and not every step is a no-brainer, that is true.
The class is not supposed to be correct, it can be touched. There are many ways to solve this thing, one possible solution might be this https://github.com/tddbin/katas/blob/b7d41a2f7307c16a71bfccd33420cccde616406b/katas/es6/language/iterator/usages.js (there is always a solved version of every kata in the repo, before I make it into a kata).
I agree that the kata could be simpler, if you have a suggestion I would be happy. PRs are welcome, so we can discuss on them. The most important thing in those katas I see is that there are learnings in the end, which build up using little steps. And I understand that everyone learns differently, its always a compromise.

Wolfram

**Comment by mattmazzola** [comment]
> The class is not supposed to be correct, it can be touched

Ah, Ok. Then it is by design then.  I see the solution you linked to addresses both the issues I mentioned but handles them in the interator function instead, which is just as effective of a solution.

I think it's safe to leave the TDDbin as is then. Thanks for responding!
